microk8s-resources/actions/common/utils.sh

```diff
@@ -219,6 +219,46 @@ skip_opt_in_config() {
     fi
 }
 
+# Function to sanitize arguments for API server
+sanatize_args_kubeapi_server() {
+  local deprecated_args=("insecure-port" "feature-gates=RemoveSelfLink")
+
+  # Set the service name
+  local service_name="kube-apiserver"
+
+  for arg in "${deprecated_args[@]}"; do
+    echo "Removing argument: $arg from $service_name"
+    skip_opt_in_local_config "$arg" "$service_name"
+  done
+}
```

Comment:
This section that is identical to all functions would better be a function of its own.